pkg/process/process.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package process
+
+import (
+	"fmt"
+
+	"github.com/prometheus/procfs"
+
+	"github.com/parca-dev/parca-agent/pkg/cgroup"
+)
+
+// key is a unique identifier for a process.
+// TODO(kakkoyun): This is not unique enough. We will need to add more fields.
```

Comment:
I think what you've done is fine since there is a case for when the start time has changed. It doesn't necessarily have to be part of the key as long as it's otherwise handled.